Sources/PWM.swift

```diff
@@ -160,33 +238,35 @@ public class RaspberryPWM: PWMOutput {
     /// The signal starts, asynchronously(manged by a device external to the CPU), once this method is called and
     /// needs to be stopped manually calling `stopPWM()`.
     public func startPWM(period ns: Int, duty percent: Float) {
-        // Kill the clock
-        clockBasePointer.advanced(by: 40).pointee = CLKM_PASSWD | CLKM_CTL_KILL     //CM CTL register: Set KILL flag
-        usleep(10)
-
-        // If the required frequency is too high, this value reduces the number of samples (scale does the opposite)
-        let highFreqSampleReduction: UInt = (ns < 750) ? 10 : 1
+        if !phy.isReadyForPwm {
+            phy.setPwmClock()
+        }
 
-        let freq: UInt = (1_000_000_000/UInt(ns)) * 100 / highFreqSampleReduction
-        let (idiv, scale) = calculateDIVI(base: .PLLD, desired: freq)                                //Using the faster (with known freq) available clock to reduce jitter
+        // @ 250Mhz, one slot in the PWM channel is 4 ns. All valid periods are multiples of that.
+        // Pick a range that is at least as large as requested that we can represent, and round the
+        // desired duty cycle to the nearest whole slot.
+        //
+        // Double precision is used here to handle very low frequencies in the range of <15Hz.
+        // Otherwise we lose precision calculating data. 
+        let range = UInt(max((Double(ns) / 4.0).rounded(.awayFromZero), 1))
```

Comment:
This part is updated from the original PR. It fixes the goofy "must have a 750ns range" behavior, and also fixes a goof on my part that I noticed where the range is assumed to be in nanoseconds, but the clock is running at 4 ns per tick. 

I'm still testing this particular change, as it means my frequencies are about 1/4th what I thought they were. I'll follow up once I am satisfied this is working the way I think it should be working.